pandapower/build_branch.py

```diff
@@ -587,30 +601,88 @@ def _calc_tap_from_dataframe(net, trafo_df):
         sin = lambda x: np.sin(np.deg2rad(x))
         arctan = lambda x: np.rad2deg(np.arctan(x))
 
-        for side, vn, direction in [("hv", vnh, 1), ("lv", vnl, -1)]:
-            phase_shifters = tap_phase_shifter & (tap_side == side)
-            tap_complex = np.isfinite(tap_step_percent) & np.isfinite(tap_pos) & (tap_side == side) & \
-                ~phase_shifters
-            if tap_complex.any():
-                tap_steps = tap_step_percent[tap_complex] * tap_diff[tap_complex] / 100
-                tap_angles = _replace_nan(tap_step_degree[tap_complex])
-                u1 = vn[tap_complex]
-                du = u1 * _replace_nan(tap_steps)
-                vn[tap_complex] = np.sqrt((u1 + du * cos(tap_angles)) ** 2 + (du * sin(tap_angles)) ** 2)
-                trafo_shift[tap_complex] += (arctan(direction * du * sin(tap_angles) /
-                                                    (u1 + du * cos(tap_angles))))
-            if phase_shifters.any():
-                degree_is_set = _replace_nan(tap_step_degree[phase_shifters]) != 0
-                percent_is_set = _replace_nan(tap_step_percent[phase_shifters]) != 0
-                if (degree_is_set & percent_is_set).any():
-                    raise UserWarning(
-                        "Both tap_step_degree and tap_step_percent set for ideal phase shifter")
-                trafo_shift[phase_shifters] += np.where(
-                    (degree_is_set),
-                    (direction * tap_diff[phase_shifters] * tap_step_degree[phase_shifters]),
-                    (direction * 2 * np.rad2deg(np.arcsin(tap_diff[phase_shifters] * \
-                                                          tap_step_percent[phase_shifters] / 100 / 2)))
-                )
+        if f'tap{t}_phase_shifter_type' in trafo_df:
+            # tap_phase_shift_type is only in dataframe starting from pp Version 3.0, older version use different logic
```

Comment:
Why not use convert_format to convert older networks?
If these backwards compatibility hacks are cluttered around the code, this will become unmaintainable in the long term.